src/actor/component.rs

```diff
@@ -20,6 +20,8 @@ pub(super) struct Ctx<'a> {
     pub wasi: ::host::WasiCtx,
     pub claims: &'a jwt::Claims<jwt::Actor>,
     pub handler: Arc<Box<dyn Handle<Invocation>>>,
+    /// Opaque context that can be supplied upon initial invocation of an actor
+    pub call_context: Option<Vec<u8>>,
```

Comment:
How about a generic

```suggestion
    pub call_context: Option<C>,
```

Where `C` defaults to `()` (`C = ()`)?

I really do not like the "opaque optional byte blob" here. What if I want to store a struct here? A number?

Reply:
I like the idea of a generic 👍🏻

Reply:
If the consumer wants to store a struct, they can serialize it themselves. Why do we need to inject monomorphic type generation for the context to hold a value it will never inspect?

If the runtime was going to inspect it or call some standardized function on it I can see the desire for generics. but the runtime never interrogates the context so it doesn't need to know its data type.

Reply:
Opting to do this later as an enhancement https://github.com/wasmCloud/wasmCloud/issues/299